packages/react-test-renderer/src/ReactTestRendererAct.js

```diff
@@ -27,9 +25,9 @@ let actingUpdatesScopeDepth = 0;
 
 function flushEffectsAndMicroTasks(onDone: (err: ?Error) => void) {
   try {
-    flushPassiveEffects();
+    Scheduler.unstable_flushWithoutYielding();
     enqueueTask(() => {
-      if (flushPassiveEffects()) {
+      if (hasPendingEffects()) {
```

Comment:
Maybe this should check if any work is scheduled, not just effects? `Scheduler.hasPendingWork()`